extensions/amp-carousel/0.1/base-slides.js

```diff
@@ -234,4 +241,19 @@ export class BaseSlides extends BaseCarousel {
     this.hasAutoplay_ = false;
     this.shouldAutoplay_ = this.hasAutoplay_ && this.isLoopingEligible();
   }
+
+  /**
+   * @protected
+   */
+  userInteractedWith() {
+    this.signals().signal(EngagementSignals.USER_INTERACTED);
```

Comment:
IIRC signals are terminal states, so this implementation assumes that autoplay is disabled after user interaction. Can we avoid coupling this to behavior and instead treat each event distinctly?

Reply:
Good suggestion--removed the use of signals.